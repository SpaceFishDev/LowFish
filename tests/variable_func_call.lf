i32 x(i32 a)
{
	return a;
}
i32 main()
{
	i32 b = 5;
	i32 n = x(a);
}
