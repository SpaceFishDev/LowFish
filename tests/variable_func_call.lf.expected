<NODE( 'PROGRAM', PROGRAM )>
{
    <NODE( 'FUNCTION', FUNCTION )>
    {
        <NODE( 'i32', TYPE )>
        <NODE( 'x', TOKEN_NODE )>
        <NODE( 'VARDECL', VARDECL )>
        {
            <NODE( 'i32', TYPE )>
            <NODE( 'a', TOKEN_NODE )>
        }
        <NODE( 'BLOCK', BLOCK )>
        {
            <NODE( 'EXPRESSION', EXPRESSION )>
            {
                <NODE( 'a', RETURN )>
            }
        }
    }
    <NODE( 'FUNCTION', FUNCTION )>
    {
        <NODE( 'i32', TYPE )>
        <NODE( 'main', TOKEN_NODE )>
        <NODE( 'BLOCK', BLOCK )>
        {
            <NODE( 'EXPRESSION', EXPRESSION )>
            {
                <NODE( 'VARDECL', VARDECL )>
                {
                    <NODE( 'i32', TYPE )>
                    <NODE( 'ASSIGNMENT', ASSIGNMENT )>
                    {
                        <NODE( 'b', TOKEN_NODE )>
                        <NODE( 'BASICEXPRESSION', BASICEXPRESSION )>
                        {
                            <NODE( '5', TOKEN_NODE )>
                        }
                    }
                }
            }
            <NODE( 'EXPRESSION', EXPRESSION )>
            {
                <NODE( 'VARDECL', VARDECL )>
                {
                    <NODE( 'i32', TYPE )>
                    <NODE( 'ASSIGNMENT', ASSIGNMENT )>
                    {
                        <NODE( 'n', TOKEN_NODE )>
                        <NODE( 'x', FUNCTION_CALL )>
                        {
                            <NODE( 'BASICEXPRESSION', BASICEXPRESSION )>
                            {
                                <NODE( 'a', TOKEN_NODE )>
                            }
                        }
                    }
                }
            }
        }
    }
}

